pkgs/by-name/ne/nexusmods-app/package.nix

```
@@ -44,44 +38,53 @@ buildDotnetModule rec {
 
   preConfigure = ''
     substituteInPlace Directory.Build.props \
-      --replace '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
+      --replace-fail '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
   '';
 
-  postPatch = ''
-    ln --force --symbolic "${lib.getExe _7zzWithOptionalUnfreeRarSupport}" src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz
-  '';
+  dotnetBuildFlags = [
+    # Tell the app it is a distro package; affects wording in update prompts
+    "--property:INSTALLATION_METHOD_FLATPAK=1"
+
+    # Don't include upstream's 7zz binary; we use the nixpkgs version
+    "--property:NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR=1"
+  ];
 
   makeWrapperArgs = [
-    "--prefix PATH : ${lib.makeBinPath [desktop-file-utils]}"
+    "--prefix PATH : ${lib.makeBinPath [ desktop-file-utils ]}"
     "--set APPIMAGE $out/bin/${meta.mainProgram}" # Make associating with nxm links work on Linux
   ];
 
+  propagatedBuildInputs = [ (_7zz.override { inherit enableUnfree; }) ];
+
   runtimeDeps = [
     fontconfig
     libICE
     libSM
     libX11
   ];
 
-  executables = [
-    nexusmods-app.meta.mainProgram
-  ];
+  executables = [ nexusmods-app.meta.mainProgram ];
```

```suggestion
  executables = [ meta.mainProgram ];
```

Since we're using a recursive attrset anyway